ddtrace/contrib/botocore/services/bedrock.py

```diff
@@ -0,0 +1,297 @@
+            )
+
+
+def _extract_request_params(params: Dict[str, Any], provider: str) -> Dict[str, Any]:
+    """
+    Extracts request parameters including prompt, temperature, top_p, max_tokens, and stop_sequences.
+    """
+    request_body = json.loads(params.get("body"))
+    if provider == _AI21:
+        return {
+            "prompt": request_body.get("prompt"),
+            "temperature": request_body.get("temperature", None),
+            "top_p": request_body.get("topP", None),
+            "max_tokens": request_body.get("maxTokens", None),
+            "stop_sequences": request_body.get("stopSequences", []),
+        }
+    elif provider == _AMAZON:
+        text_generation_config = request_body.get("textGenerationConfig", {})
+        return {
+            "prompt": request_body.get("inputText"),
+            "temperature": text_generation_config.get("temperature", None),
+            "top_p": text_generation_config.get("topP", None),
+            "max_tokens": text_generation_config.get("maxTokenCount", None),
+            "stop_sequences": text_generation_config.get("stopSequences", []),
+        }
+    elif provider == _ANTHROPIC:
+        return {
+            "prompt": request_body.get("prompt"),
+            "temperature": request_body.get("temperature", None),
+            "top_p": request_body.get("top_p", None),
+            "top_k": request_body.get("top_k", None),
+            "max_tokens": request_body.get("max_tokens_to_sample", None),
+            "stop_sequences": request_body.get("stop_sequences", []),
+        }
+    elif provider == _COHERE:
+        return {
+            "prompt": request_body.get("prompt"),
+            "temperature": request_body.get("temperature", None),
+            "top_p": request_body.get("p", None),
+            "top_k": request_body.get("k", None),
+            "max_tokens": request_body.get("max_tokens", None),
+            "stop_sequences": request_body.get("stop_sequences", []),
+            "stream": request_body.get("stream", None),
+            "n": request_body.get("num_generations", None),
+        }
+    elif provider == _META:
+        return {
+            "prompt": request_body.get("prompt"),
+            "temperature": request_body.get("temperature", None),
+            "top_p": request_body.get("top_p", None),
+            "max_tokens": request_body.get("max_gen_len", None),
+        }
+    elif provider == _STABILITY:
+        # TODO: request/response formats are different for image-based models. Defer for now
+        return {}
+    return {}
+
+
+def _extract_response(span: Span, body: Dict[str, Any]) -> Dict[str, List[str]]:
+    """
+    Extracts text and finish_reason from the response body, which has different formats for different providers.
+    """
+    text, finish_reason = None, None
+    provider = span.get_tag("bedrock.request.model_provider")
+    if provider == _AI21:
+        text = body.get("completions")[0].get("data").get("text")
```

Comment:
Are completions and results guaranteed to have at least one element? Should we check the size of the element first?

Reply:
yup, the results are guaranteed to be this format. What I can do instead is to add error handling in case of index/attribute error, and return None on error.